src/gt4py/next/iterator/type_inference.py

```diff
@@ -475,108 +590,163 @@ def visit_Lambda(
         ret = self.visit(node.expr, **kwargs)
         return FunctionType(args=Tuple.from_elems(*(ptypes[p.id] for p in node.params)), ret=ret)
 
-    def visit_FunCall(
-        self,
-        node: ir.FunCall,
-        **kwargs,
-    ) -> Type:
-        if isinstance(node.fun, ir.SymRef):
-            if node.fun.id == "make_tuple":
-                # Calls to make_tuple are handled as being part of the grammar,
-                # not as function calls
-                argtypes = self.visit(node.args, **kwargs)
-                kind = TypeVar.fresh()
-                size = TypeVar.fresh()
-                dtype = Tuple.from_elems(*(TypeVar.fresh() for _ in argtypes))
-                for d, a in zip(dtype, argtypes):
-                    self.constraints.add((Val(kind=kind, dtype=d, size=size), a))
-                return Val(kind=kind, dtype=dtype, size=size)
-            if node.fun.id == "tuple_get":
-                # Calls to tuple_get are handled as being part of the grammar,
-                # not as function calls
-                if len(node.args) != 2:
-                    raise TypeError("tuple_get requires exactly two arguments")
-                if not isinstance(node.args[0], ir.Literal) or node.args[0].type != "int":
-                    raise TypeError("The first argument to tuple_get must be a literal int")
-                idx = int(node.args[0].value)
-                tup = self.visit(node.args[1], **kwargs)
-                kind = TypeVar.fresh()
-                elem = TypeVar.fresh()
-                size = TypeVar.fresh()
-
-                dtype = Tuple(front=elem, others=TypeVar.fresh())
-                for _ in range(idx):
-                    dtype = Tuple(front=TypeVar.fresh(), others=dtype)
-
-                val = Val(
-                    kind=kind,
-                    dtype=dtype,
+    def _visit_make_tuple(self, node: ir.FunCall, **kwargs) -> Type:
+        # Calls to make_tuple are handled as being part of the grammar,
+        # not as function calls
+        argtypes = self.visit(node.args, **kwargs)
+        kind = (
+            TypeVar.fresh()
+        )  # `kind == Iterator()` means zipping iterators into an iterator of tuples
+        size = TypeVar.fresh()
+        dtype = Tuple.from_elems(*(TypeVar.fresh() for _ in argtypes))
+        for d, a in zip(dtype, argtypes):
+            self.constraints.add((Val(kind=kind, dtype=d, size=size), a))
+        return Val(kind=kind, dtype=dtype, size=size)
+
+    def _visit_tuple_get(self, node: ir.FunCall, **kwargs) -> Type:
+        # Calls to tuple_get are handled as being part of the grammar,
+        # not as function calls
```

Comment:
```suggestion
        # Calls to `tuple_get` are handled as being part of the grammar, not as function calls.
```
line length + backticks.